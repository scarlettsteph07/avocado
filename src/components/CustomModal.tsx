import React, { useState } from 'react'
import Modal from 'react-modal'

import styled from '../styled'
import { StyledIconButton } from './styled'

type Props = {
  children: JSX.Element
}

export const CustomModal: React.FunctionComponent<Props> = ({
  children,
}: Props) => {
  Modal.setAppElement('#root')
  const [isModalOpen, setIsModalOpen] = useState(false)

  const openModal = (): void => {
    setIsModalOpen(true)
  }

  const closeModal = (): void => {
    setIsModalOpen(false)
  }

  const customStyles = {
    content: {
      width: '70%',
      top: '50%',
      left: '50%',
      right: 'auto',
      bottom: 'auto',
      marginRight: '-50%',
      transform: 'translate(-50%, -50%)',
      display: 'flex',
      flexFlow: 'column',
      border: '2px solid #202020',
      borderRadius: '3px',
    },
  }

  const childrenWithProps = React.Children.map(children, (child) =>
    React.cloneElement(child, { closeModal: closeModal }),
  )

  return (
    <CustomModalStyles>
      <OpenIconButton onClick={openModal} />
      <Modal
        contentLabel="Modal"
        isOpen={isModalOpen}
        onRequestClose={closeModal}
        style={customStyles}
      >
        <CloseIconButton onClick={closeModal} />
        {childrenWithProps}
      </Modal>
    </CustomModalStyles>
  )
}

const OpenIconButton = styled(StyledIconButton)`
  margin: 0;
  background-image: url(/svg/icon--plus.svg);
  align-self: flex-end;
`

const CloseIconButton = styled(StyledIconButton)`
  margin: 0;
  background-image: url(/svg/icon--x.svg);
  align-self: flex-end;
`
const CustomModalStyles = styled.div``
