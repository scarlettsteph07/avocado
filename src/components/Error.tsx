import React from 'react'
import _ from 'lodash'

import styled from '../styled'
import { StyledSlatOuter, StyledSlatInner } from './styled/'

type Props = { error: string }

export const Error: React.FunctionComponent<Props> = ({
  error,
}: Props) => {
  return (
    <ErrorStyles className="error">
      <StyledSlatOuter className="error__outer">
        <StyledSlatInner className="error__inner">
          <h1>Oops, something went wrong!</h1>
          <h3>{_.startCase(error)}</h3>
        </StyledSlatInner>
      </StyledSlatOuter>
    </ErrorStyles>
  )
}

const ErrorStyles = styled.div``
