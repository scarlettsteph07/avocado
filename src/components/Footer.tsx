import React from 'react'

import styled from '../styled'

interface Props {

}

export const Footer = () => {
  const currentTime = new Date()
  return (
    <FooterStyles className='footer'>
      &copy; {currentTime.getFullYear()} scarlettsteph07. All Rights Reserved.
    </FooterStyles>
  )
}

const FooterStyles = styled.footer`
  height: 25px;
  display: flex;
  justify-content: center;
  align-items: center;
  @media (min-width: ${({ theme }) => theme.medium.start} ) {
    height: 70px;
  }
`