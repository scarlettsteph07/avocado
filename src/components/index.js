export * from './Body'
export * from './Footer'
export * from './Header'
export * from './IngredientsGrid'
export * from './IngredientsList'
export * from './IngredientPage'
export * from './Picker'
export * from './Recipe'
export * from './RecipeContainer'
export * from './Settings'
export * from './SubHeader'
