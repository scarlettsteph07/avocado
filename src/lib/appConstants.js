export const API_URL = 'https://5nzohvug5m.execute-api.us-east-1.amazonaws.com/dev/ingredients'