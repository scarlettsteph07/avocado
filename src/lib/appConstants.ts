import { Payload } from '../types'

export const API_URL = 'https://api.toastandavocado.xyz/v1/recipes'

export const API_DEV_USER = 'webapp-dev'

export const API = {
  ALL: {
    URL: 'https://api.toastandavocado.xyz/v1/ingredients',
    METHOD: 'GET',
  },
  NEW: {
    URL: 'https://api.toastandavocado.xyz/v1/ingredients/new',
    METHOD: 'POST',
  },
  DELETE: {
    URL: 'https://api.toastandavocado.xyz/v1/ingredients',
    METHOD: 'DELETE',
  },
  NEW_STYLE: {
    URL: 'https://api.toastandavocado.xyz/v1/ingredients/styles',
    METHOD: 'POST',
  },
  UPDATE_STYLE: {
    URL: 'https://api.toastandavocado.xyz/v1/ingredients/styles',
    METHOD: 'PUT',
  },
}

export const APPLICATION_PATHS = {
  INGREDIENTS: 'ingredients',
}

export const APP_TITLE = 'Toast & Avocado'

export const DEFAULT_PAYLOAD: Payload = {
  dietPreference: 'carnivore',
  ignoredIngredients: [],
  numOfOptionalIngredients: 3,
  requestedIngredients: [],
}

export const DIET_PREFERENCES = ['carnivore', 'vegan', 'vegetarian']
