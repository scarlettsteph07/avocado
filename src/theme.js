export let theme = {
  small: {
    outer: '320px',
    inner: '320px',
    start: '0px',
    end: '640px',
  },
  medium: {
    inner: '640px',
    start: '641px',
    end: '1024px'
  },
  large: {
    before: '1139px',
    inner: '1140px',
    start: '1141px',
    end: '1440px',
  },
  xlarge: {
    outer: '1440px',
    inner: '1140px',
    start: '1441px'
  },
  devices: {
    desktop: '926px',
    tablet: '768px',
  },

  baseFont: '"Scarlett", "Annie Use Your Telescope", cursive',
  secondaryFont: '"Helvetica Neue", Helvetica, sans-serif',

  textPrimary:              '#202020',
  transparent:              'transparent',
  transparentOverlay:       'rgba(0,0,0,0.2)',
  semiTransparentOverlay:   'rgba(0,0,0,0.05)',
  black:                    '#000000',
  white:                    '#ffffff',
  emerald:                  '#4ed164',
  lightGray:                '#D9DADC',
  red:                      '#FF0000',
}
