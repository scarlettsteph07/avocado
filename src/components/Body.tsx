import React from 'react'
import axios from 'axios'

import styled from '../styled'
import { API_URL } from '../lib/appConstants'
import {
  IngredientsList,
  Settings,
} from './'
import { StyledButton } from './styled/'

import { Ingredient, Payload } from '../types/'

interface Props {

}

interface State {
  isLoading: boolean,
  ingredients: Array<Ingredient>,
  payload: Payload,
}

class Body extends React.Component<Props, State> {
  state: State = {
    isLoading: false,
    ingredients: [],
    payload: {
      isCarnivore: false,
      dietPreference: 'carnivore',
      numOfOptionalIngredients: 1,
    }
  }

  componentDidMount = () => {
    this.setState({ isLoading: true })
    this.fetchIngredients(this.setPayload())
  }

  setPayload = (): Payload => {
    const { payload } = this.state
    return {
      "isCarnivore": payload.isCarnivore,
      "dietPreference": payload.dietPreference,
      "numOfOptionalIngredients": payload.numOfOptionalIngredients,
    }
  }

  fetchIngredients = (data: Object) => {
    axios.post(
      API_URL,
      data,
    )
    .then( response => {
      const { ingredients } = response.data
      this.setState({ ingredients, isLoading: false })
    })
    .catch(error => {
      console.log(error)
    })
  }

  handleOnButtonClick = () => {
    this.fetchIngredients(this.setPayload())
  }

  updateIsCarnivore = () => {
    this.setState(prevState => ({
      payload: {
        ...prevState.payload,
        isCarnivore: !this.state.payload.isCarnivore,
      }
    }), () => this.fetchIngredients(this.setPayload()))
  }

  updateNumberOfIngredients = (numOfOptionalIngredients: number) => {
    this.setState(prevState => ({
      payload: {
        ...prevState.payload,
        numOfOptionalIngredients,
      }
    }), () => this.fetchIngredients(this.setPayload()))
  }

  render() {
    const { isLoading, payload } = this.state
    return (
    <BodyStyles className="body">
      <Settings
        updateIsCarnivore={this.updateIsCarnivore}
        isCarnivore={payload.isCarnivore}
        dietPreference={payload.dietPreference}
        updateNumberOfIngredients={this.updateNumberOfIngredients}
        numOfOptionalIngredients={payload.numOfOptionalIngredients}
      />
      <IngredientsList ingredients={this.state.ingredients} />
      { !isLoading &&
        <RandomizeButton
          className="randomize-button"
          onClick={(this.handleOnButtonClick)}
        >
          Randomize!
        </RandomizeButton>
      }
    </BodyStyles>
  )}
}

export { Body }

const BodyStyles = styled.div`
  width: 100%;
  min-height: 90vh;
  box-sizing: border-box;
  font-size: 25px;
  color: ${({ theme }) => theme.textPrimary};
  background: white;
  background: linear-gradient(0, ${({theme}) => theme.snowDrift} 8%, ${({theme}) => theme.oldPaper} 8%) 0 57px;
  background-size: 100% 30px;
  display: flex;
  flex-flow: column;
  align-items: center;
  justify-content: flex-start;
  @media (min-width: ${({ theme }) => theme.medium.start} ) {
    font-size: 70px;
  }
`

const RandomizeButton = styled(StyledButton)`
  min-width: 25%;
  height: auto;
  margin: 25px auto;
  font-size: 25px;
  @media (min-width: ${({ theme }) => theme.medium.start} ) {
    font-size: 50px;
  }
`