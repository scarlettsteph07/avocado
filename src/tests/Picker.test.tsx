import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'emotion-theming'

import { DIET_PREFERENCES } from '../lib/appConstants'
import { Picker } from '../components/'
import { theme } from '../theme'

describe('Picker component', () => {
  test('should match snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <Picker
          dropdownTitle="diet preference"
          onChange={() => {}}
          value="carnivore"
          wheelsData={DIET_PREFERENCES}
        />
      </ThemeProvider>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
