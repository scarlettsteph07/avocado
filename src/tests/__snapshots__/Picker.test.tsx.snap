// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker component should match snapshot 1`] = `
<div
  className="css-vhwtt1"
  id="diet-preference"
>
  carnivore
</div>
`;
